Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs

```diff
@@ -296,31 +312,44 @@ private CSProjectInfo GetProjectInfo(Dictionary<string, CSProjectInfo> projectsM
             return toReturn;
         }
 
-        private List<PluginAssemblyInfo> ScanForPluginDLLs()
+        private void ScanForReferences(out List<PluginAssemblyInfo> plugins, out List<WinMDInfo> winmds)
         {
-            List<PluginAssemblyInfo> toReturn = new List<PluginAssemblyInfo>();
+            plugins = new List<PluginAssemblyInfo>();
+            winmds = new List<WinMDInfo>();
 
-            foreach (string assetAssemblyPath in Directory.GetFiles(Utilities.AssetPath, "*.dll", SearchOption.AllDirectories))
+            IEnumerable<string> assetReferences = Directory.GetFiles(Utilities.AssetPath, "*.*", SearchOption.AllDirectories)
+                .Where(file => file.ToLower().EndsWith(".dll") || file.ToLower().EndsWith(".winmd"));
```

Comment:
>file.ToLower() [](start = 31, length = 14)

Could do .Select(t=>t.ToLower()).Where(t=> t.EndsWith() || ...)

Reply:
if you go this route you end up with all lower case file names which can cause issues with finding the asset relative path

Reply:
Ah, yes, my bad. Was thinking of how best to cache. Could you pass StringCompareOptions to EndsWith?